spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/SleuthKafkaStreamsConfiguration.java

```diff
@@ -0,0 +1,124 @@
+ */
+
+package org.springframework.cloud.sleuth.instrument.messaging;
+
+import java.util.Map;
+
+import brave.Tracing;
+import brave.kafka.clients.KafkaTracing;
+import brave.kafka.streams.KafkaStreamsTracing;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.streams.processor.internals.DefaultKafkaClientSupplier;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.sleuth.instrument.messaging.TraceMessagingAutoConfiguration.SleuthKafkaConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.config.KafkaStreamsConfiguration;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration
+ * Auto-configuration} enables Kafka Streams span creation and reporting.
+ *
+ * @author Tim te Beek
+ */
+@Configuration
+@ConditionalOnBean({ Tracing.class, KafkaStreamsConfiguration.class })
+@ConditionalOnProperty(value = "spring.sleuth.messaging.kafka.streams.enabled", matchIfMissing = true)
+@AutoConfigureAfter({ SleuthKafkaConfiguration.class })
+public class SleuthKafkaStreamsConfiguration {
+
+	protected SleuthKafkaStreamsConfiguration() {
+	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	KafkaStreamsTracing kafkaStreamsTracing(Tracing tracing) {
+		return KafkaStreamsTracing.create(tracing);
+	}
+
+	@Bean
+	static KafkaStreamsBuilderFactoryBeanPostProcessor kafkaStreamsBuilderFactoryBeanPostProcessor(
+			KafkaTracing tracing) {
+		return new KafkaStreamsBuilderFactoryBeanPostProcessor(tracing);
+	}
+
+}
+
+class KafkaStreamsBuilderFactoryBeanPostProcessor implements BeanPostProcessor {
+
+	private static final Log log = LogFactory
+			.getLog(KafkaStreamsBuilderFactoryBeanPostProcessor.class);
+
+	private final KafkaTracing kafkaTracing;
+
+	KafkaStreamsBuilderFactoryBeanPostProcessor(KafkaTracing kafkaTracing) {
+		this.kafkaTracing = kafkaTracing;
+	}
+
+	@Override
+	public Object postProcessAfterInitialization(Object bean, String beanName)
+			throws BeansException {
+		if (bean instanceof StreamsBuilderFactoryBean) {
+			StreamsBuilderFactoryBean sbfb = (StreamsBuilderFactoryBean) bean;
+			if (log.isDebugEnabled()) {
+				log.debug("StreamsBuilderFactoryBean bean is auto-configured to enable tracing.");
+			}
+			sbfb.setClientSupplier(new SleuthKafkaClientSupplier(kafkaTracing));
```

Comment:
Use `KafkaStreamsTracing.kafkaClientSupplier()` through https://github.com/apache/incubator-zipkin-brave/pull/887/files#diff-b8dd8e4445870a6a249cf383bdfd6cf7R54 once Brave 5.6.4 is used.